### PR TITLE
Use one(T), zero(T) instead of 1, 0 in Julia implementation

### PR DESCRIPTION
**Pull request by Tortar**
added one(T) and zero(T) instead of 1 and 0, it was already done inside one of the loops where it counts more for performance, here I added it more for consistency, but it should be (almost **un**noticeably) faster

### CODE REVIEW

**Comment by Moelf** [comment]
That one is actually needed if I remember correctly

File: julia/Related/src/Related.jl [hunk 1]
```diff
@@ -20,9 +20,9 @@ end
 
 StructTypes.StructType(::Type{PostData}) = StructTypes.Struct()
 
-function fastmaxindex!(xs::Vector, topn, maxn, maxv)
-    maxn .= 1
-    maxv .= 0
+function fastmaxindex!(xs::Vector, maxn, maxv, ::Type{T}) where {T}
```
Comment (Moelf) [inline julia/Related/src/Related.jl:23]: Why do you need an extra argument? Just use eltype of maxn

File: julia/Related/src/Related.jl [hunk 2]
```diff
@@ -45,10 +45,9 @@ end
 function related(posts)
     Ts = (UInt8, UInt16, UInt32, UInt64)
     i = findfirst(T -> length(posts) < typemax(T), Ts)
-    return related(Ts[i], posts)
+    return related(posts, Ts[i])
 end
-function related(::Type{T}, posts) where {T}
-    topn = 5
+function related(posts, ::Type{T}) where {T}
```
Comment (Moelf) [inline julia/Related/src/Related.jl:50]: What's the point of this?

**Comment by Moelf** [comment]
Our local testing sometimes miss the CI machines performance characteristics, please stop micro adjusting it for style or something else

**Comment by Moelf** [comment]
For example: https://github.com/jinyus/related_post_gen/pull/182

File: julia/Related/src/Related.jl (hunk 1 above)
Comment (Moelf) [reply julia/Related/src/Related.jl:23]: Or type maxn with `::AbstractVector{T}` and use the T in function body

**Review by Moelf — APPROVED** [review]

**Comment by Tortar** [comment]
wow, you are totally right, I rerun the main branch vs. this PR (before last commit) and saw a regression, sorry for that, strangely enough I didn't see it the first time I ran the benchmarks, now I took extra care to benchmark and it should be fine. I think I learnt what you said in the other PR :D

**Pull request merged**